admin-dev/themes/new-theme/js/app/utils/init-components.js

```diff
@@ -37,6 +37,50 @@ import TextWithLengthCounter from '@components/form/text-with-length-counter';
 import PreviewOpener from '@components/form/preview-opener';
 import MultistoreConfigField from '@js/components/form/multistore-config-field.js';
 import {EventEmitter} from '@components/event-emitter';
+import GridComponent from '@components/grid/grid';
+import Router from '@components/router';
+
+// Grid extensions
+import AsyncToggleColumnExtension from '@components/grid/extension/column/common/async-toggle-column-extension';
+import BulkActionCheckboxExtension from '@components/grid/extension/bulk-action-checkbox-extension';
+import BulkOpenTabsExtension from '@components/grid/extension/bulk-open-tabs-extension';
+import ChoiceExtension from '@components/grid/extension/choice-extension';
+import ColumnTogglingExtension from '@components/grid/extension/column-toggling-extension';
+import ExportToSqlManagerExtension from '@components/grid/extension/export-to-sql-manager-extension';
+import FiltersResetExtension from '@components/grid/extension/filters-reset-extension.js';
+import FiltersSubmitButtonEnablerExtension from '@components/grid/extension/filters-submit-button-enabler-extension';
+import LinkRowActionExtension from '@components/grid/extension/link-row-action-extension';
+import ModalFormSubmitExtension from '@components/grid/extension/modal-form-submit-extension';
+import PositionExtension from '@components/grid/extension/position-extension';
+import PreviewExtension from '@components/grid/extension/preview-extension';
+import ReloadListExtension from '@components/grid/extension/reload-list-extension';
+import SortingExtension from '@components/grid/extension/sorting-extension';
+import SubmitBulkActionExtension from '@components/grid/extension/submit-bulk-action-extension';
+import SubmitGridActionExtension from '@components/grid/extension/submit-grid-action-extension';
+import SubmitRowActionExtension from '@components/grid/extension/action/row/submit-row-action-extension';
+
+const Grid = {
+  core: GridComponent,
```

Comment:
I'm not sure about this naming which can be confusing, this means in the module you'd need to do something like:

```js
const grid = new window.prestashop.components.Grid.core();
```

Which feels a bit off, I'm not sure about the naming but maybe?

```suggestion
  GridComponent: GridComponent,
```

This would turn the module code into:
```
const grid = new window.prestashop.components.Grid.GridComponent();
```

Or maybe we actually should split this so we could do:

```
const grid = new window.prestashop.components.Grid();
grid.addExtension(new window.prestashop.components.gridExtensions.BulkActionCheckboxExtension());
```

Reply:
I also wonder what would happen if this component is used in the `initComponents` method, since it's usually used like this:
```js
initComponents([
   'TranslatableField',
   'TranslatableInput',
   'Grid',
]);
```

But the grid object needs additional parameters so it won't work I guess, should we handle a special case to avoid it being created via the `initComponents` function? (I know people are not supposed to use it this way but we can't expect this to never be done ^^)

Reply:
I quite like this idea:

```
const grid = new window.prestashop.components.Grid();
grid.addExtension(new window.prestashop.components.gridExtensions.BulkActionCheckboxExtension());
```

what do you think @NeOMakinG ?

Reply:
Yep looks fine and you're right @jolelievre, didn't think about the instantiation use with this naming